tests/components/shelly/test_button.py

```diff
@@ -0,0 +1,70 @@
+"""Tests for Shelly button platform."""
+from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN
+from homeassistant.components.button.const import SERVICE_PRESS
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_registry import async_get
+
+
+async def test_block_button(hass: HomeAssistant, coap_wrapper):
+    """Test block device OTA button."""
+    assert coap_wrapper
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(coap_wrapper.entry, BUTTON_DOMAIN)
+    )
+    await hass.async_block_till_done()
+
+    # stable channel button
+    state = hass.states.get("button.test_name_ota_update")
+    assert state
+    assert state.state == STATE_UNKNOWN
+
+    await hass.services.async_call(
+        BUTTON_DOMAIN,
+        SERVICE_PRESS,
+        {ATTR_ENTITY_ID: "button.test_name_ota_update"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    coap_wrapper.device.trigger_ota_update.assert_called_once_with(beta=False)
+
+    # beta channel button
+    entity_registry = async_get(hass)
+    entry = entity_registry.async_get("button.test_name_ota_update_beta")
+    state = hass.states.get("button.test_name_ota_update_beta")
+
+    assert entry
+    assert state is None
```

Comment:
After pressing the button, the state should be the time the button was pressed, not None. There seems to be something wrong?

Reply:
it is not the `state.state` ... it is just the check that the beta button is not enabled by default

Reply:
> is not enabled by default, so not present in the state machine
In that case, I'm confused, why can it be called by the above service call in that case? If the button entity isn't enabled, that shouldn't be working...

Reply:
the service call is for `button.test_name_ota_update` (without beta prefix) ;)

Reply:
![](https://media2.giphy.com/media/26xBQSVMbdZCRctri/giphy.gif)

Reply:
it's ok ... the weekend is near ... take a beer and relax 🍻